conference/react-js/src/containers/SignInForm/index.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { PureComponent } from 'react';
+
+class SignInForm extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      isValidCredentials: true,
+    };
+
+    this.emailChangeHandler = this.emailChangeHandler.bind(this);
+    this.submitHandler = this.submitHandler.bind(this);
+    this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
+  }
+
+  submitHandler(event) {
```

Comment:
To avoid binding context for functions in the constructor, use such definition: 
```
submitHandler = event = {
   this.setState()
   ...
}
```

As You know, an arrow function always has a context where it was created.

Reply:
I made common handler to form change event.
    this.formChangeHandler = (event) => {
      const target = event.target;

      this.setState({
        [target.name]: target.value,
        isValidCredentials: true,
      });
    };

Is it ok?